flink-table/flink-sql-parser/src/main/codegen/includes/parserImpls.ftl

```diff
@@ -574,13 +575,18 @@ SqlAlterTable SqlAlterTable() :
                         startPos.plus(getPos()));
         }
     |
-        <DROP> <CONSTRAINT>
-        constraintName = SimpleIdentifier() {
-            return new SqlAlterTableDropConstraint(
-                tableIdentifier,
-                constraintName,
-                startPos.plus(getPos()));
-        }
+        <DROP>
+        (
+            <CONSTRAINT> constraintName = SimpleIdentifier() {
+                return new SqlAlterTableDropConstraint(
+                            tableIdentifier,
+                            constraintName,
+                            startPos.plus(getPos()));
+            }
+        |
+            [ <IF> <EXISTS> { ifExists = true; } ]
```

Comment:
Should it be `<IF> <EXISTS> <PARTITION>`? what if a sql statement like `alter table drop table`? Seems it will also fall back to `SqlDropPartitions`.
Maybe it's a little confused that any `drop` except for `CONSTRAINT` will fall back to `SqlDropPartitions`. It may bring some complexity when try to support other `drop`.